src/main/java/com/achdev/onlinebookstoreapp/exception/InstanceCreationException.java

```diff
@@ -0,0 +1,7 @@
+package com.achdev.onlinebookstoreapp.exception;
+
+public class InstanceCreationException extends RuntimeException {
```

Comment:
Where did you got this exception?

Reply:
private <T> T createInstance(Class<T> clazz) {
        try {
            return clazz.getDeclaredConstructor().newInstance();
        } catch (Exception e) {
            throw new InstanceCreationException("Failed to create instance of class: "
                                                + clazz.getName());
        }
    }